jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/annotations/AnnotationDiagnosticsCollector.java

```diff
@@ -1,273 +1,299 @@
+                        for (ILocalVariable parameter : parameters) {
+                        	annotations = parameter.getAnnotations();
+                            for (IAnnotation annotation : annotations) {
+                            	if (isValidAnnotation(annotation.getElementName(), validAnnotations))
+                            		annotatables.add(new Tuple.Two<>(annotation, parameter));
+                            }                        	
+                        }
+                    }
+                    // Field
+                    IField[] fields = type.getFields();
+                    for (IField field : fields) {
+                    	annotations = field.getAnnotations();
+                        for (IAnnotation annotation : annotations) {
+                        	if (isValidAnnotation(annotation.getElementName(), validTypeAnnotations))
+                        		annotatables.add(new Tuple.Two<>(annotation, field));
+                        }
+                    }
+                }
+
+                for (Tuple.Two<IAnnotation, IAnnotatable> annotatable : annotatables) {
+                    IAnnotation annotation = annotatable.getFirst();
+                    IAnnotatable element = annotatable.getSecond();
+
+                    if (isMatchedAnnotation(unit, annotation, AnnotationConstants.GENERATED_FQ_NAME)) {
+                        for (IMemberValuePair pair : annotation.getMemberValuePairs()) {
+                            // If date element exists and is non-empty, it must follow ISO 8601 format.
+                            if (pair.getMemberName().equals("date")) {
+                                if (pair.getValue() instanceof String) {
+                                    String date = (String) pair.getValue();
+                                    if (!date.equals("")) {
+                                        if (!Pattern.matches(AnnotationConstants.ISO_8601_REGEX, date)) {
+                                            ISourceRange annotationNameRange = JDTUtils.getNameRange(annotation);
+
+                                            Range annotationRange = JDTUtils.toRange(
+                                                    unit,
+                                                    annotationNameRange.getOffset(),
+                                                    annotationNameRange.getLength());
+                                            String diagnosticMessage = generateDiagnosticAnnotation("Generated","define the attribute 'date' following the ISO 8601 standard.");
+
+                                            Diagnostic diagnostic = new Diagnostic(
+                                                    annotationRange,
+                                                    diagnosticMessage);
+                                            completeDiagnostic(diagnostic, AnnotationConstants.DIAGNOSTIC_CODE_DATE_FORMAT);
+                                            diagnostics.add(diagnostic);

```

Comment:
This code is duplicated a few times in this class - should we use the new `createDiagnostic` method in `AbstractDiagnosticsCollector` that both creates and completes the diagnostic instead

Reply:
Yes, the class was my first change at the time, then kept adding new util. APIs, I will clean it up in the future PR(s), my have to change/add new APIs